### DISCUSSION TITLE
MySQL Boolean datatype

### DISCUSSION THREAD

**Discussion by adamjehu** [Budibase/budibase#2589] [Help] [answered]
Hi, I'm connected to a MySQL database which I'm using as my main data table. I have two Boolean Fields - however Budibase reads them as numerical (since they are). Is there any way to set the Boolean datatype for external tables in Budibase? This has become necessary since I would like to use a checkbox to update these fields. I'm new to Budibase and would appreciate any help with this.

**Comment by mjashanks** [comment, marked as answer]
Hi @adamjehu . At the moment, Budibase will convert `Boolean` (checkbox) to `tinyint` (MySql column) automatically. So, checkbox should work with `tinyint`.

If you only want to handle an `update` or `insert`, you can use the `Save Row` action to override the value that gets sent to your database. I.e. you can transform `true` into `1` and `false` into `0`.

The video below shows how you can do this with a handlebars expression. Unfortunately there will be no way to read `1` from the database, and set the checkbox to `true`. It's a good shout though - we should handle this better! Thanks!


https://user-images.githubusercontent.com/3524181/133141624-d7e46689-ebe9-46b0-9b43-c2498889df74.mp4

**Reply by adamjehu** [reply]
Hello @mjashanks. Thank you for your response. I followed the video and added the checkbox under Field Group, but I cannot select the corresponding field in its settings. Works fine once the field type is set to boolean for internal tables - but not with the MySQL tables it seems.

**Reply by mjashanks** [reply]
Hey @adamjehu - you don't need to select a field - you can free type into the "Field" input. Call it anything you like - I'm my example, I called it "IsWorking".

Then, in the `Save Row` action, you can use the field name that you typed.

**Reply by adamjehu** [reply]
Thank you for all your assistance. Apart from it pulling the field value to be 1, it works great!
